tests/conftest.py

```diff
@@ -1,9 +1,86 @@
+import random
+import time
+import warnings
+from pathlib import Path
 from unittest.mock import MagicMock
 
 import pytest
 from click.testing import CliRunner
+from freezegun import freeze_time
 
-from abdiff.core import init_job
+from abdiff.core import init_job, init_run
+from abdiff.core.utils import create_subdirectories
+
+
+class Container:
+    """Stub for docker.models.container.Container object."""
+
+    def __init__(self, id, labels):  # noqa: A002
+        self.id = id
+        self.labels = labels
+        self.status = "created"
+
+    def reload(self):
+        time.sleep(0.1)
+        if random.randint(0, 100) > 50:  # noqa: PLR2004, S311
+            self.status = "exited"
+
+    def logs(self):
+        with open("tests/fixtures/transmogrifier-logs.txt", "rb") as file:
+            return file.read()
+
+
+class MockedContainerRun:
+    """Class for mocking container runs of Transmogrifier."""
+
+    def __init__(self) -> None:
+        self.count = 0
+
+    def yield_mocked_run(self, transformed_directories: tuple[str]) -> None:
+        """Perform a mocked run of transmogrifier.
+
+        This function yields a different outcome based on the value of
+        MockedContainerRun.count each time it is called. The function will
+        create a placeholder transformed file in the A/B transformed
+        directories in the correct order (first in 'transformed/a' then
+        in 'transformed/b').
+        """
+        transformed_directory_a, transformed_directory_b = transformed_directories
+        if self.count == 0:
+            self.count += 1
```

Comment:
Very minor, but `self.count += 1` could be bumped outside of the `if/elif` checks, then bump both of those by 1, e.g.

```python
self.count += 1
if self.count == 1:  # first time called
  # do stuff
elif self.count == 2: # second time called
  # do stuff
else:
  # ruh-roh, shouldn't be here
```

Reply:
Change has been made.

Reply:
See [updated `MockedContainerRun.yield_mocked_run`](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/6/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R49).